src/lib.rs

```diff
@@ -162,6 +171,60 @@ impl Client {
     /// ```
     /// let client = influxdb2::Client::new("http://localhost:8888", "org", "my-token");
     /// ```
+    pub fn new(
+        url: impl Into<String>,
+        org: impl Into<String>,
+        auth_token: impl Into<String>,
+    ) -> Self {
+        // unwrap is used to maintain backwards compatibility
+        // panicking was present earlier as well inside of reqwest
+        ClientBuilder::new(url, org, auth_token).build().unwrap()
+    }
+
+    /// Consolidate common request building code
+    fn request(&self, method: Method, url: &str) -> reqwest::RequestBuilder {
+        let mut req = self.reqwest.request(method, url);
+
+        if let Some(auth) = &self.auth_header {
+            req = req.header("Authorization", auth.expose_secret());
+        }
+
+        req
+    }
+
+    /// Join base Url of the client to target API endpoint into valid Url
+    fn url(&self, endpoint: &str) -> String {
+        let mut url = self.base.clone();
+        url.set_path(endpoint);
+        url.into()
+    }
+}
+
+/// Errors that occur when building the client
+#[derive(Debug, Snafu)]
+pub enum BuildError {
+    /// While constructing the reqwest client an error occured
+    #[snafu(display("Error while building the client: {}", source))]
+    ReqwestClientError {
+        /// Reqwest internal error
+        source: reqwest::Error,
+    },
+}
+/// ClientBuilder builds the `Client`
+#[derive(Debug)]
+pub struct ClientBuilder {
```

Comment:
`ClientBuilder` was added to be able to enable/disable compression on the reqwest client through the reqwest `ClientBuilder`

Reply:
Love the client builder pattern

Reply:
The pattern was lifted from reqwest, so credit goes to them. In fact I wouldn't have been able to figure out that I had to use `StreamReader` and `ReaderStream` without looking at their gzip implementation for responses